portfolio/src/main/webapp/script.js

```diff
@@ -12,25 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-
-let index = 0;
+/*
+    The function showNext() allows the button to be clicked twice in order to
+    display a personal summary and a photograph. Because the second click is dependent
+    on the first, a global variable is required to determine when to show the picture.
+    Another roundabout solution would be to check whether the 'summaryCointainer'
+    is empty, eliminating the need for a global varible.
```

Comment:
I'm glad you're thinking about alternate solutions. In this case there's no need to mention the alternate approach in the source code (it would likely be in a design doc instead, for a large project). Some cases where you would mention alternate approaches in the source:
- If there's a more straightforward approach, which you can't use for some non-obvious reason. Then it's good to explain why you're taking the roundabout approach.
- If there's a better approach (e.g. a faster algorithm), but you haven't implemented it yet (e.g. because it's much more complex or it's not yet clear that the algorithm is a performance bottleneck in real usage).